[GLIB] Update apt dependency list

dpino opened the pull request:
#### e352042b6251531e8a1af8f303b0943ccad219c6
<pre>
[GLIB] Update apt dependency list
<a href="https://bugs.webkit.org/show_bug.cgi?id=279504">https://bugs.webkit.org/show_bug.cgi?id=279504</a>

Reviewed by Adrian Perez de Castro.

Upgrade list of Debian/Ubuntu packages required to build WebKitGTK and
WPE. Some packages are no longer needed, so they were removed. Other
packages are always guaranteed to exist, so there&apos;s no need to install
and alternative package.

* Tools/glib/dependencies/apt:
* Tools/gtk/dependencies/apt:
* Tools/wpe/dependencies/apt:

Canonical link: <a href="https://commits.webkit.org/283537@main">https://commits.webkit.org/283537@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0716993b46fcd80d730b6d7800c023f38d1f33c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66402 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45777 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19023 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70435 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17053 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68520 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53576 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17295 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53258 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11872 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69469 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42198 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57481 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33911 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38869 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14869 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15888 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60766 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15211 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72138 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10359 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14585 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60582 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10391 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57550 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60896 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8552 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2162 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10096 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41584 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42661 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43844 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42404 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Tools/glib/dependencies/apt:
```diff
@@ -36,50 +36,49 @@ PACKAGES=(
     mesa-common-dev
     ninja-build
     patch
+    patchelf
```
I highly doubt WebKit itself depends on `patchelf`. What would be this used for?

dpino replied:
That's correct, `patchelf` is not required to build WebKitGTK or WPE, but there are several scripts in `Tools/Scripts`, such as `Tools/Scripts/bundle-binary` and `Tools/Scripts/generate-bundle`, that use `patchelf`.

In addition, `patchelf` is also bundled in the FlatpakSDK (`Tools/buildstream/elements/sdk/patchelf.bst`), so it makes sense to me that it's part of the list of OS packages.

dpino replied:
For instance, consider the bot [GTK-Linux-64bit-Release-Packaging-Nightly-Ubuntu2004](https://build.webkit.org/#/builders/30). This bot is used for generating the nightly binaries featured in [wpt.fyi](https://wpt.fyi/results/?label=experimental&label=master&aligned). This bot builds with system libraries plus JHBuild (`minimal-and-gstreamer` moduleset).

This bot also includes a step [generate-minibrowser-bundle](https://build.webkit.org/#/builders/30/builds/9094/steps/11/logs/stdio), that uses the script `Tools/Script/generate-bundle`. This bot is working because someone manually installed `patchelf` in the bot. If we deploy this bot from scratch again, it won't work since `patchelf` is needed.

aperezdc reviewed (commented):
Please check my question below.

aperezdc reviewed (approved):
Thanks for the carification @dpino, r=me.

dpino committed:
[GLIB] Update apt dependency list
https://bugs.webkit.org/show_bug.cgi?id=279504

Reviewed by Adrian Perez de Castro.

Upgrade list of Debian/Ubuntu packages required to build WebKitGTK and
WPE. Some packages are no longer needed, so they were removed. Other
packages are always guaranteed to exist, so there's no need to install
and alternative package.

* Tools/glib/dependencies/apt:
* Tools/gtk/dependencies/apt:
* Tools/wpe/dependencies/apt:

Canonical link: https://commits.webkit.org/283537@main
